provider/reprovider.go

```diff
@@ -424,7 +424,7 @@ func (s *reprovider) run() {
 				return
 			}
 
-			err := s.reprovide(s.ctx, false)
+			err := s.reprovide(context.Background(), false)
```

Comment:
This is done because `s.reprovide` is already selecting over `s.ctx` so there is no additional context to watch when calling `s.reprovide`